clients/src/main/java/org/apache/kafka/clients/consumer/internals/AbstractCoordinator.java

```diff
@@ -723,12 +732,20 @@ protected synchronized void markCoordinatorUnknown(boolean isDisconnected) {
      * Get the current generation state if the group is stable.
      * @return the current generation or null if the group is unjoined/rebalancing
      */
-    protected synchronized Generation generation() {
+    protected synchronized Generation generationOnGroupStable() {
         if (this.state != MemberState.STABLE)
             return null;
         return generation;
     }
 
+    /**
+     * Return generation info regardless of the group state. Only using in unit tests.
+     * @return the current generation info
+     */
+    protected synchronized Generation generation() {
```

Comment:
Can we drop to default access? I don't think sub-classes will want to touch this.